R/create.R

```diff
@@ -27,18 +32,20 @@ create_clean_reported_cases <- function(reported_cases, horizon, zero_threshold
   reported_cases <- reported_cases[is.na(breakpoint), breakpoint := 0]
   reported_cases <- data.table::setorder(reported_cases, date)
   ## Filter out 0 reported cases from the beginning of the data
-  reported_cases <- reported_cases[order(date)][
-    ,
-    cum_cases := cumsum(confirm)
-  ][cum_cases > 0][, cum_cases := NULL]
+  if (filter_leading_zeros){
+    reported_cases <- reported_cases[order(date)][
+      ,
+      cum_cases := cumsum(confirm)
+    ][cum_cases > 0][, cum_cases := NULL]    
+  }
 
-  # Check case counts surrounding zero cases and set to 7 day average if average is over 7 days
-  # is greater than a threshold
-  if (is.infinite(zero_threshold)) {
+  # Check case counts preceding zero case counts and set to 7 day average if
+  # average over last 7 days is greater than a threshold
+  if (!is.infinite(zero_threshold)) {
     reported_cases <-
       reported_cases[
         ,
-        `:=`(average_7 = data.table::frollsum(confirm, n = 8) / 7)
+        `:=`(average_7 = (data.table::frollsum(confirm, n = 8)) / 7)
```

Comment:
I've reverted this as actually not a bug. This is because it is only used when `confirm = 0` and so both definitions are the same. I appreciate this is a little opaque but keeping for the the very slight efficiency gain.